### PR TITLE
Add FXIOS-10252 Show notifications for Close Remote Tab requests from other clients

### PR DESCRIPTION
**Pull request by skhamis**
## :scroll: Tickets
[Jira ticket](https://mozilla-hub.atlassian.net/browse/FXIOS-10252)
[Github issue](https://github.com/mozilla-mobile/firefox-ios/issues/TODO)

## :pencil: Checklist
You have to check all boxes before merging
- [ ] Filled in the above information (tickets numbers and description of your work)
- [ ] Updated the PR name to follow our [PR naming guidelines](https://github.com/mozilla-mobile/firefox-ios/wiki/Pull-Request-Naming-Guide)
- [ ] Wrote unit tests and/or ensured the tests suite is passing
- [ ] When working on UI, I checked and implemented accessibility (minimum Dynamic Text and VoiceOver)
- [ ] If needed, I updated documentation / comments for complex code and public methods
- [ ] If needed, added a backport comment (example `@Mergifyio backport release/v120`)

### CODE REVIEW

File: firefox-ios/Client/Frontend/Library/HistoryPanel/HistoryPanel.swift [hunk 1]
```diff
@@ -350,6 +350,10 @@ class HistoryPanel: UIViewController,
             }
 
             showClearRecentHistory()
+        case .OpenRecentlyClosedTabs:
+            // TODO: Need to also have the history panel show up if not active
+            historyCoordinatorDelegate?.showRecentlyClosedTab()
```
Comment (skhamis) [inline firefox-ios/Client/Frontend/Library/HistoryPanel/HistoryPanel.swift:355]: So while this does work, it doesn't open up the panel...I tried timeboxing figuring out an easier way to deal with this but might need some help from the iOS team on this!

Comment (mattreaganmozilla) [reply firefox-ios/Client/Frontend/Library/HistoryPanel/HistoryPanel.swift:355]: @skhamis Just to make sure I'm understanding: is the history panel not opening at all? What happens if you manually open it (Settings > History), then close it, then attempt to run the test? 

The `HistoryPanel` isn't actually initialized until the first time the UI is opened (which typically happens in `LibraryCoordinator.makeChildPanels()`, so it's possible that there isn't an active instance of it which can respond to this notification.

If the above steps (manually opening the history UI, then closing, then testing the notification) fix the problem then it's very likely due to it simply not having been initialized yet.

Comment (skhamis) [reply firefox-ios/Client/Frontend/Library/HistoryPanel/HistoryPanel.swift:355]: The history panel is not opening at all. Even opening it -> closing won't open it up. However when I manually go to Settings -> History _after_ tapping the notification, it's already on the recently closed tab page. So I think the routing is working but just not opening the panel!

Comment (mattreaganmozilla) [reply firefox-ios/Client/Frontend/Library/HistoryPanel/HistoryPanel.swift:355]: @skhamis So I could be misunderstanding, but I suspect that the issue might be that we need to push this UI from a point higher up in the coordinator hierarchy, possibly via something like `BrowserCoordinator.showLibrary(with: .history)` and then after that we can ask the history panel itself to push to the Recently Closed Tabs.

Comment (skhamis) [reply firefox-ios/Client/Frontend/Library/HistoryPanel/HistoryPanel.swift:355]: > possibly via something like BrowserCoordinator.showLibrary(with: .history) and then after that we can ask the history panel itself to push to the Recently Closed Tabs.

Thanks @mattreaganmozilla I think I was looking for something like that! I think it's in a good place now with the panel showing up and recently closed tabs also showing when the notification is tapped!

File: firefox-ios/Client/Frontend/Strings.swift [hunk 2]
```diff
@@ -3725,6 +3725,20 @@ extension String {
         tableName: nil,
         value: "View",
         comment: "Label for an action used to view one or more tabs from a notification.")
+
+// MARK: - Close tab notifications
+    public static let CloseTab_ArrivingNotification_title = MZLocalizedString(
+        key: "CloseTab_ArrivingNotification_title",
+        tableName: "nil",
```
Comment (thatswinnie) [inline firefox-ios/Client/Frontend/Strings.swift:3732]: All new strings should have a table name. Could you please add one for both strings?

File: firefox-ios/Extensions/NotificationService/NotificationService.swift [hunk 3]
```diff
@@ -214,6 +219,16 @@ class SyncDataDisplay {
         }
     }
 
+    func displayClosedTabNotification(urls: [String]) {
+        notificationContent.userInfo[NotificationCloseTabs.closeTabsKey]
+            = [NotificationCloseTabs.messageIdKey: "closeRemoteTab"]
+        notificationContent.categoryIdentifier = NotificationCloseTabs.notificationCategoryId
+            presentNotification(
```
Comment (thatswinnie) [inline firefox-ios/Extensions/NotificationService/NotificationService.swift:226]: The indentation looks off here.

File: firefox-ios/Client/Frontend/Strings.swift (hunk 2 above)
Comment (skhamis) [reply firefox-ios/Client/Frontend/Strings.swift]: Done!

File: firefox-ios/Extensions/NotificationService/NotificationService.swift (hunk 3 above)
Comment (skhamis) [reply firefox-ios/Extensions/NotificationService/NotificationService.swift]: Good catch!

**Review by thatswinnie — APPROVED** [review]

**Comment by nbhasin2** [comment]
@thatswinnie going to merge it into main as everything looks good but should this go into v132 (its a bit close) but we are building RC on monday

**Pull request merged**